implementation/src/main/java/io/smallrye/config/TracedConfigSource.java

```diff
@@ -0,0 +1,5 @@
+package io.smallrye.config;
+
+public interface TracedConfigSource {
+    String getTrace(final String name);
```

Comment:
I don't like that getting a configuration property with its information entails a minimum of two lookups.

Reply:
We don't have to, once we have accessors directly in the ConfigSource, but this was just an easy way to do it.

Alternative, we can turn this into a wrapper around a ConfigSource, so we can control the getValue method and perform only a single lookup.